### PR TITLE
Add path when registering styles

### LINKED ISSUE #13315: Assets: Support inline styles
## Feature Description

WordPress supports inline styling using the `wp_maybe_inline_styles`. This loop around all styles and if the property of path is present, it will inline the style, saving a http request. Adding support for this in this plugin would be simple. I have provide a code snippet blow.

### PR DESCRIPTION
**Pull request by swissspidy**
## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #13315

### CODE REVIEW

File: tests/phpunit/integration/tests/Assets.php [hunk 1]
```diff
@@ -188,6 +188,19 @@ public function test_register_style_asset(): void {
...
+		 *
+		 * @var array<string, _WP_Dependency> $registered
+		 */
+		$registered = wp_styles()->registered;
+		
+		$test_style_metadata     = $registered['test_style']->extra;
+		$fake_css_chunk_metadata = $registered['fake_css_chunk']->extra;
+		
+		$this->assertArrayHasKey( 'path', $test_style_metadata );
+		$this->assertArrayHasKey( 'path', $fake_css_chunk_metadata );
```
Comment (AnuragVasanwala) [inline tests/phpunit/integration/tests/Assets.php:203]: I have updated integration test to make sure `path` is present.

**Review by AnuragVasanwala — APPROVED** [review]
@swissspidy LGTM ✅

**Comment by swissspidy** [comment]
@AnuragVasanwala Sweet, thank you so much for adding tests! 🚀

File: includes/Assets.php [hunk 2]
```diff
@@ -216,22 +217,25 @@ public function register_style_asset( string $style_handle, array $style_depende
...
 		$this->register_style(
 			$style_handle,
-			$base_style_path . $style_handle . '.css',
+			$base_style_url . $style_handle . '.css',
 			$style_dependencies,
 			$entry_version
 		);
 
 		wp_style_add_data( $style_handle, 'rtl', 'replace' );
+		wp_style_add_data( $style_handle, 'path', $base_style_path . $style_handle . '.css' );
```
Comment (spacedmonkey) [inline includes/Assets.php:238]: @swissspidy You have to manually change the path if `is_rtl()` is true.

Comment (swissspidy) [reply includes/Assets.php:238]: 1e66368e8715d082b1b6e58a8a704a960f88e49a

**Pull request merged**